server/lib/chat/chat-service.ts

```diff
@@ -200,18 +206,16 @@ export default class ChatService {
     return userSockets
   }
 
-  // TODO(tec27): type event properly
-  private publishToChannel(channelName: string, data: any) {
+  private publishToChannel<T extends ChatEvent>(channelName: string, data: T) {
     this.nydus.publish(getChannelPath(channelName), data)
   }
 
   private subscribeUserToChannel(userSockets: UserSocketsGroup, channelName: string) {
-    userSockets.subscribe(getChannelPath(channelName), () => {
-      return {
-        action: 'init',
-        activeUsers: this.state.channels.get(channelName),
-      }
-    })
+    const initEventData: ChatInitEvent = {
```

Comment:
You changed the semantics of this. Before, every time a client connected for this user, it would retrieve the active users for the channel.

With this change, it retrieves the active users for the channel *when this user was first subscribed* and never updates it again. Please revert